op.c

```diff
@@ -2069,16 +2070,69 @@ Perl_scalar(pTHX_ OP *o)
                         )
                 )
                 {
+
+                    if (OP_TYPE_IS(o, OP_LIST) && !op_parent(o)) {
+                        /* Is the list now just an obvious scalar pushop?
+                         *     <@> list sKP ->6
+                         *         <0> pushmark v ->4
+                         *         <$> const(IV 3) s ->5
+                         */
+                        OP* first = cLISTOPo->op_first;
+                        assert(OP_TYPE_IS(first, OP_PUSHMARK));
+                        OP* sib1 = OpSIBLING(first);
+                        assert(sib1);
+                        OP* sib2 = OpSIBLING(sib1);
+                        if (!sib2) {
+                            if (
+                                PL_opargs[sib1->op_type] & OA_RETSCALAR
```

Comment:
Rather than two nested ifs with no extra code, this might be neater and avoid a level of indent as `if (!sib2 && (PL_op_args[sib1->op_type] & OA_RETSCALAR))`